addons/microsoft_calendar/models/calendar.py

```diff
@@ -117,14 +156,19 @@ def write(self, values):
                 e._microsoft_delete(e._get_organizer(), e.ms_organizer_event_id, timeout=3)
                 e.microsoft_id = False
 
-        notify_context = self.env.context.get('dont_notify', False)
         res = super(Meeting, self.with_context(dont_notify=notify_context)).write(values)
 
         if recurrence_update_setting in ('all_events',) and len(self) == 1 \
            and values.keys() & self._get_microsoft_synced_fields():
             self.recurrence_id.need_sync_m = True
         return res
 
+    def action_mass_archive(self, recurrence_update_setting):
+        # Do not allow archiving if recurrence is synced with Outlook. Suggest updating directly from Outlook.
+        if self._check_microsoft_sync_status() and any(self.microsoft_id):
+            self._forbid_recurrence_update()
+        super().action_mass_archive(recurrence_update_setting)
+
     def _get_microsoft_sync_domain(self):
         # in case of full sync, limit to a range of 1y in past and 1y in the future by default
         ICP = self.env['ir.config_parameter'].sudo()
```

Comment:
Note: This comment is not valid anymore. 
# Do not sync events that follow the recurrence, they are already synced at recurrence creation